### ISSUE TITLE
Is this training process correct?

### ISSUE THREAD

**Issue by coolKeen** [pkuxmq/Invertible-Image-Rescaling#9]
@pkuxmq  Hi, when I run "python train.py -opt  options/train/train_IRN_x4.yml" (nothing change in train_IRN_x4.yml) . I find the validation log as below:

20-07-28 20:02:43.941 - INFO: <epoch: 99, iter:   5,000> psnr: 2.9862e+01.
20-07-28 21:41:00.651 - INFO: <epoch:199, iter:  10,000> psnr: 3.0981e+01.
20-07-28 23:24:36.609 - INFO: <epoch:299, iter:  15,000> psnr: 3.1322e+01.
20-07-29 01:05:20.143 - INFO: <epoch:399, iter:  20,000> psnr: 3.1410e+01.
20-07-29 02:43:22.314 - INFO: <epoch:499, iter:  25,000> psnr: 3.1553e+01.
20-07-29 04:24:01.338 - INFO: <epoch:599, iter:  30,000> psnr: 3.1646e+01.
20-07-29 06:05:05.640 - INFO: <epoch:699, iter:  35,000> psnr: 3.1639e+01.
20-07-29 07:43:24.878 - INFO: <epoch:799, iter:  40,000> psnr: 3.1862e+01.
20-07-29 09:25:29.029 - INFO: <epoch:899, iter:  45,000> psnr: 3.1765e+01.
20-07-29 11:04:49.718 - INFO: <epoch:999, iter:  50,000> psnr: 3.1857e+01.
20-07-29 12:43:52.355 - INFO: <epoch:1099, iter:  55,000> psnr: 3.1875e+01.
20-07-29 14:26:15.819 - INFO: <epoch:1199, iter:  60,000> psnr: 3.2005e+01.
20-07-29 16:06:06.029 - INFO: <epoch:1299, iter:  65,000> psnr: 3.1971e+01.
20-07-29 17:43:06.634 - INFO: <epoch:1399, iter:  70,000> psnr: 3.1794e+01.
20-07-29 19:25:28.171 - INFO: <epoch:1499, iter:  75,000> psnr: 3.1746e+01.
20-07-29 21:08:17.830 - INFO: <epoch:1599, iter:  80,000> psnr: 3.1740e+01.
20-07-29 22:46:15.390 - INFO: <epoch:1699, iter:  85,000> psnr: 3.1913e+01.
20-07-30 00:26:51.227 - INFO: <epoch:1799, iter:  90,000> psnr: 3.1833e+01.
20-07-30 02:07:26.182 - INFO: <epoch:1899, iter:  95,000> psnr: 3.1923e+01.
20-07-30 03:45:40.531 - INFO: <epoch:1999, iter: 100,000> psnr: 3.1870e+01.
20-07-30 05:26:20.864 - INFO: <epoch:2099, iter: 105,000> psnr: 3.2184e+01.
20-07-30 07:09:07.773 - INFO: <epoch:2199, iter: 110,000> psnr: 3.2158e+01.
20-07-30 08:51:11.258 - INFO: <epoch:2299, iter: 115,000> psnr: 3.2238e+01.
20-07-30 10:34:42.669 - INFO: <epoch:2399, iter: 120,000> psnr: 3.2157e+01.
20-07-30 12:16:32.068 - INFO: <epoch:2499, iter: 125,000> psnr: 3.2186e+01.
20-07-30 14:05:44.933 - INFO: <epoch:2599, iter: 130,000> psnr: 3.2131e+01.
20-07-30 16:04:55.918 - INFO: <epoch:2699, iter: 135,000> psnr: 3.2209e+01.
20-07-30 18:02:46.877 - INFO: <epoch:2799, iter: 140,000> psnr: 3.2193e+01.
20-07-30 20:01:30.929 - INFO: <epoch:2899, iter: 145,000> psnr: 3.2184e+01.
20-07-30 22:00:33.235 - INFO: <epoch:2999, iter: 150,000> psnr: 3.2234e+01.
20-07-30 23:58:48.030 - INFO: <epoch:3099, iter: 155,000> psnr: 3.2176e+01.
20-07-31 01:58:16.605 - INFO: <epoch:3199, iter: 160,000> psnr: 3.2134e+01.
20-07-31 03:57:55.036 - INFO: <epoch:3299, iter: 165,000> psnr: 3.2190e+01.
20-07-31 05:57:29.178 - INFO: <epoch:3399, iter: 170,000> psnr: 3.2055e+01.
20-07-31 08:34:08.977 - INFO: <epoch:3499, iter: 175,000> psnr: 3.2192e+01.
20-07-31 10:34:28.891 - INFO: <epoch:3599, iter: 180,000> psnr: 3.2137e+01.
20-07-31 12:32:34.701 - INFO: <epoch:3699, iter: 185,000> psnr: 3.2155e+01.
20-07-31 14:30:21.658 - INFO: <epoch:3799, iter: 190,000> psnr: 3.2176e+01.
20-07-31 16:30:04.046 - INFO: <epoch:3899, iter: 195,000> psnr: 3.2132e+01.
20-07-31 18:27:40.744 - INFO: <epoch:3999, iter: 200,000> psnr: 3.2186e+01.
20-07-31 21:18:03.124 - INFO: <epoch:4099, iter: 205,000> psnr: 3.2356e+01.
20-07-31 23:18:25.399 - INFO: <epoch:4199, iter: 210,000> psnr: 3.2415e+01.
20-08-01 01:30:43.122 - INFO: <epoch:4299, iter: 215,000> psnr: 3.2399e+01.
20-08-01 03:29:23.416 - INFO: <epoch:4399, iter: 220,000> psnr: 3.2401e+01.
20-08-01 05:42:45.864 - INFO: <epoch:4499, iter: 225,000> psnr: 3.2354e+01.
20-08-01 07:41:49.838 - INFO: <epoch:4599, iter: 230,000> psnr: 3.2396e+01.
20-08-01 09:40:05.609 - INFO: <epoch:4699, iter: 235,000> psnr: 3.2428e+01.
20-08-01 11:35:44.124 - INFO: <epoch:4799, iter: 240,000> psnr: 3.2334e+01.
20-08-01 13:25:11.782 - INFO: <epoch:4899, iter: 245,000> psnr: 3.2112e+01.
20-08-01 14:40:20.978 - INFO: <epoch:4999, iter: 250,000> psnr: 3.2427e+01.
20-08-01 15:34:56.752 - INFO: <epoch:5099, iter: 255,000> psnr: 3.2367e+01.
20-08-01 16:31:03.010 - INFO: <epoch:5199, iter: 260,000> psnr: 3.2382e+01.
20-08-01 17:26:51.330 - INFO: <epoch:5299, iter: 265,000> psnr: 3.2443e+01.
20-08-01 18:30:16.764 - INFO: <epoch:5399, iter: 270,000> psnr: 3.2142e+01.
20-08-01 19:40:53.347 - INFO: <epoch:5499, iter: 275,000> psnr: 3.2391e+01.
20-08-01 20:35:14.469 - INFO: <epoch:5599, iter: 280,000> psnr: 3.2404e+01.
20-08-01 21:54:01.785 - INFO: <epoch:5699, iter: 285,000> psnr: 3.2371e+01.
20-08-01 23:15:16.187 - INFO: <epoch:5799, iter: 290,000> psnr: 3.2415e+01.
20-08-02 00:37:40.440 - INFO: <epoch:5899, iter: 295,000> psnr: 3.2385e+01.
20-08-02 01:35:00.178 - INFO: <epoch:5999, iter: 300,000> psnr: 3.2393e+01.
20-08-02 02:42:40.792 - INFO: <epoch:6099, iter: 305,000> psnr: 3.2570e+01.
20-08-02 04:02:45.011 - INFO: <epoch:6199, iter: 310,000> psnr: 3.2594e+01.
20-08-02 05:21:02.174 - INFO: <epoch:6299, iter: 315,000> psnr: 3.2575e+01.
20-08-02 06:41:36.365 - INFO: <epoch:6399, iter: 320,000> psnr: 3.2598e+01.
20-08-02 08:01:57.685 - INFO: <epoch:6499, iter: 325,000> psnr: 3.2608e+01.
20-08-02 09:21:45.858 - INFO: <epoch:6599, iter: 330,000> psnr: 3.2603e+01.
20-08-02 10:43:11.029 - INFO: <epoch:6699, iter: 335,000> psnr: 3.2582e+01.
20-08-02 12:04:38.429 - INFO: <epoch:6799, iter: 340,000> psnr: 3.2589e+01.
20-08-02 13:27:13.231 - INFO: <epoch:6899, iter: 345,000> psnr: 3.2610e+01.
20-08-02 14:48:32.692 - INFO: <epoch:6999, iter: 350,000> psnr: 3.2611e+01.
20-08-02 16:08:36.940 - INFO: <epoch:7099, iter: 355,000> psnr: 3.2601e+01.
20-08-02 17:29:17.291 - INFO: <epoch:7199, iter: 360,000> psnr: 3.2577e+01.
20-08-02 18:49:37.341 - INFO: <epoch:7299, iter: 365,000> psnr: 3.2599e+01.
20-08-02 20:10:23.640 - INFO: <epoch:7399, iter: 370,000> psnr: 3.2640e+01.
20-08-02 21:36:42.104 - INFO: <epoch:7499, iter: 375,000> psnr: 3.2617e+01.
20-08-02 23:10:36.419 - INFO: <epoch:7599, iter: 380,000> psnr: 3.2585e+01.
20-08-03 00:50:44.942 - INFO: <epoch:7699, iter: 385,000> psnr: 3.2614e+01.
20-08-03 02:30:23.124 - INFO: <epoch:7799, iter: 390,000> psnr: 3.2633e+01.
20-08-03 04:08:43.651 - INFO: <epoch:7899, iter: 395,000> psnr: 3.2592e+01.
20-08-03 05:48:41.351 - INFO: <epoch:7999, iter: 400,000> psnr: 3.2608e+01.
20-08-03 07:28:49.655 - INFO: <epoch:8099, iter: 405,000> psnr: 3.2696e+01.
20-08-03 09:08:23.989 - INFO: <epoch:8199, iter: 410,000> psnr: 3.2598e+01.
20-08-03 10:50:37.364 - INFO: <epoch:8299, iter: 415,000> psnr: 3.2601e+01.
20-08-03 12:32:00.880 - INFO: <epoch:8399, iter: 420,000> psnr: 3.2716e+01.
20-08-03 14:11:19.063 - INFO: <epoch:8499, iter: 425,000> psnr: 3.2717e+01.

I find that in your paper (table 1), the psnr on the validation set of DIV2K is 35.07 (500k iteration). But the psnr in the validation log  above is around 32.7 (425k iteration).  It may be something wrong happen?

Environmet:
pytorch 1.2.0
cuda 10.2
4x Tesla v100 (32GB)

**Comment by pkuxmq** [comment]
Hi, it is correct. The results in the paper are PSNR and SSIM on the Y channel of images represented in the YCbCr color space, which follows the setting in other works. The validation results in the training code are PSNR on RGB three-channel images, so they are usually lower, which is the same with other works. In test.py, for a trained model, PSNR on both RGB channels and Y channel will be calculated.

**Comment by coolKeen** [comment]
> Hi, it is correct. The results in the paper are PSNR and SSIM on the Y channel of images represented in the YCbCr color space, which follows the setting in other works. The validation results in the training code are PSNR on RGB three-channel images, so they are usually lower, which is the same with other works. In test.py, for a trained model, PSNR on both RGB channels and Y channel will be calculated.

OK, I see. Thanks!
